lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/evaluator/VortexWorker.java

```diff
@@ -97,30 +99,26 @@ public void run() {
           }
 
           // Command Executor: Deserialize the command
-          final VortexRequest vortexRequest = vortexAvroUtils.toVortexRequest(message);
+          final MasterToWorkerRequest masterToWorkerRequest = (MasterToWorkerRequest)kryoUtils.deserialize(message);
 
-          switch (vortexRequest.getType()) {
+          switch (masterToWorkerRequest.getType()) {
             case AggregateTasklets:
-              final TaskletAggregationRequest taskletAggregationRequest = (TaskletAggregationRequest) vortexRequest;
+              final TaskletAggregationRequest taskletAggregationRequest =
+                  (TaskletAggregationRequest) masterToWorkerRequest;
               aggregates.put(taskletAggregationRequest.getAggregateFunctionId(),
-                  new AggregateContainer(heartBeatTriggerManager, vortexAvroUtils, workerReports,
+                  new AggregateContainer(heartBeatTriggerManager, kryoUtils, workerReports,
                       taskletAggregationRequest));
-
-              // VortexFunctions need to be put into the repository such that VortexAvroUtils will know how to
-              // convert inputs and functions into a VortexRequest on subsequent messages requesting to
-              // execute the aggregateable tasklets.
               aggregateFunctionRepository.put(taskletAggregationRequest.getAggregateFunctionId(),
```

Comment:
Now that we don't need to encode/decode with a Codec, I think we can actually remove `aggregateFunctionRepository` from `VortexWorker`.